aws-api/src/main/java/com/amplifyframework/api/aws/auth/JWTTokenApiRequestSigner.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.aws.auth;
+
+import com.amazonaws.Request;
+
+import java.util.function.Supplier;
```

Comment:
The `java.util.function.*` are from Java 8, and potentially may be desugared by core library desugaring?

A 100% safe alternative would just be to add an inner interface:
```java
public class JWTTokenApiRequestSigner extends ApiRequestSigner {
    public JWTTokenApiRequestSigner(TokenSupplier tokenSupplier) {
        ...
    }
    ...
    @FunctionalInterface public interface TokenSupplier {
        public String supply();
    }
    ...
}
```

Reply:
That's right...forgot about that. Every time I use `Supplier` I think to myself "why don't I see more of it in the codebase?" 🤦  🤣